.github/scripts/action.yaml

```diff
@@ -0,0 +1,51 @@
+name: Copy library generation script
+description: Copy library generation shell script to the calling workflow
+inputs:
+  repo_full_name:
+    description: full name of the repository in which the head branch lives
+    required: true
+  github_repository:
+    description: the full name of the repository in which base branch lives
+    required: true
+  base_ref:
+    description: base branch
+    required: true
+  head_ref:
+    description: head branch
+    required: true
+  token:
+    description: PAT
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+  - name: Prerequisite
+    shell: bash
+    run: |
+      if [[ "${GITHUB_REPOSITORY}" != "${REPO_FULL_NAME}" ]]; then
+        echo "This PR comes from a fork. Skip library generation."
+        exit 0
```

Comment:
Do you think we should fail this step?

Reply:
I think any way of stopping the workflow without throwing an error would work fine. I added this comment because I think this `exit 0` will be considered as a successful step and [the next step would then start](https://docs.github.com/en/actions/sharing-automations/creating-actions/setting-exit-codes-for-actions#about-exit-codes). Maybe using `GITHUB_OUTPUT` along with `if` in the following steps may do the job ([SO](https://stackoverflow.com/a/61832535/7435861))?

Reply:
I added a separate job to determine whether to run the library generation.

If the pull request comes from a fork, the generation will be skipped.